hal/interfaces/I2c.hpp

```diff
@@ -54,6 +54,33 @@ namespace hal
         virtual void ReceiveData(I2cAddress address, infra::ByteRange data, Action nextAction, infra::Function<void(Result)> onReceived) = 0;
     };
 
+    class I2cErrorPolicy
+    {
+    protected:
+        I2cErrorPolicy() = default;
+        I2cErrorPolicy(const I2cErrorPolicy& other) = delete;
+        I2cErrorPolicy& operator=(const I2cErrorPolicy& other) = delete;
+        ~I2cErrorPolicy() = default;
+
+    public:
+        virtual void DeviceNotFound() = 0;
+        virtual void BusError() = 0;
+        virtual void ArbitrationLost() = 0;
+    };
+
+    class I2cMasterWithErrorPolicy : public I2cMaster
```

Comment:
The SetErrorPolicy and ResetErrorPolicy can be added to I2cMaster, since it's low impact, and a non-intrusive but core part of I2C interactions